File: handroid/src/main/java/com/h/android/http/BaseUrlProvider.java
package com.h.android.http;

import java.lang.annotation.ElementType;

/**
 * 2020/11/20
 *
 * @author zhangxiaohui
 * @describe
 */
@java.lang.annotation.Target({ElementType.TYPE})
@java.lang.annotation.Retention(java.lang.annotation.RetentionPolicy.RUNTIME)
@java.lang.annotation.Documented
@java.lang.annotation.Inherited
public @interface BaseUrlProvider {

    Class<? extends UrlProvider> value();
}

File: handroid/src/main/java/com/h/android/http/annotation/BaseUrlProvider.kt
package com.h.android.http.annotation

import com.h.android.http.UrlProvider
import java.lang.annotation.Inherited
import kotlin.reflect.KClass

/**
 * 2020/12/5
 *绑定域名/主要路由,优先级高
 * @author zhangxiaohui
 * @describe
 */
@kotlin.annotation.Target(AnnotationTarget.CLASS)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Inherited
annotation class BaseUrlProvider(val value: KClass<out UrlProvider>)